x-pack/plugins/apm/public/components/app/settings/agent_explorer/index.tsx

```diff
@@ -0,0 +1,200 @@
+              <h2>
+                {i18n.translate('xpack.apm.settings.agentExplorer.title', {
+                  defaultMessage: 'Agent explorer',
+                })}
+              </h2>
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <TechnicalPreviewBadge icon="beaker" />
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiTitle>
+      </EuiFlexItem>
+      <EuiSpacer />
+      <EuiFlexItem grow={false}>
+        <KueryBar />
+      </EuiFlexItem>
+      <EuiSpacer />
+      <EuiFlexItem>
+        <EuiFlexGroup justifyContent="flexEnd">
+          <EuiFlexItem grow={false}>
+            <SuggestionsSelect
+              prepend={i18n.translate(
+                'xpack.apm.agentExplorer.serviceNameSelect.label',
+                {
+                  defaultMessage: 'Service name',
+                }
+              )}
+              defaultValue={serviceName}
+              fieldName={SERVICE_NAME}
+              onChange={(value) => {
+                urlHelpers.push(history, {
+                  query: { serviceName: value ?? '' },
+                });
+              }}
+              placeholder={i18n.translate(
+                'xpack.apm.agentExplorer.serviceNameSelect.placeholder',
+                {
+                  defaultMessage: 'All',
+                }
+              )}
+              start={start}
+              end={end}
+              dataTestSubj="agentExplorerServiceNameSelect"
+            />
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <SuggestionsSelect
+              prepend={i18n.translate(
+                'xpack.apm.agentExplorer.agentLanguageSelect.label',
+                {
+                  defaultMessage: 'Agent language',
+                }
+              )}
+              defaultValue={agentLanguage}
+              fieldName={SERVICE_LANGUAGE_NAME}
+              onChange={(value) => {
+                urlHelpers.push(history, {
+                  query: { agentLanguage: value ?? '' },
+                });
+              }}
+              placeholder={i18n.translate(
+                'xpack.apm.agentExplorer.agentLanguageSelect.placeholder',
+                {
+                  defaultMessage: 'All',
+                }
+              )}
+              start={start}
+              end={end}
+              dataTestSubj="agentExplorerAgentLanguageSelect"
+            />
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiFlexItem>
+      <EuiFlexItem>
+        <EuiCallOut
+          size="s"
+          title={i18n.translate('xpack.apm.agentExplorer.callout.24hoursData', {
+            defaultMessage: 'Information based on the lastest 24h',
+          })}
+          iconType="clock"
+        />
+      </EuiFlexItem>
+      <EuiSpacer />
+      <EuiFlexItem>
+        <AgentList
+          isLoading={isLoading}
+          items={(agents.data?.items ?? []).filter(
+            (agent) => agent.instances > 0
```

Comment:
Why do we need to filter here?

Reply:
There are services that have no instances, is that right?